network/p2p/p2p.go

```diff
@@ -173,3 +183,23 @@ func (s *serviceImpl) Conns() []network.Conn {
 func (s *serviceImpl) ClosePeer(peer peer.ID) error {
 	return s.host.Network().ClosePeer(peer)
 }
+
+// netAddressToListenAddress converts a netAddress in "ip:port" format to a listen address
+// that can be passed in to libp2p.ListenAddrStrings
+func netAddressToListenAddress(netAddress string) (string, error) {
```

Comment:
Identical to https://github.com/algorand/go-algorand/pull/5710/files#diff-e188dd59048c59cf62e08ce53b9cccd49d50c20772824c846ee06d146228bec8R57

Put this in the peerstore package so we can refactor that function to use this one if you want to merge this separately.

Reply:
They are actually semantically different. `:8080` is a valid netAddress for these purposes and would be used in the config where as the `PeerInfoFromDomainPort` is meant to be used with DHT or DNS results in host:port format and there the port validation wouldn't be valid.

This function also uses IP `/ip4/` instead of the `/dns4/`. I'm happy to merge them if necessary.

I don't see a lot of value in refactoring them but if you have a preferred approach I'm happy to do it. Also if you just want me to move it to `peerstore/utils.go` as is can do that as well but this is unlikely to be used in context of the PeerStore since allowing passing in a port only is not a valid thing in that context outside of private networks running on the same host.

Reply:
Ah good point, I hadn't noticed that difference!